File: ui/src/models/api/note.ts
import { NoteType } from './enums/noteType';

export default interface Note {
    id: number;
    title: string;
    content: string;
    preview: string;
    noteType: NoteType;
    notebookId: number;
    created: Date;
    changed: Date;
    opened: Date;
}

File: src/Ducode.Wolk.Domain/Entities/Note.cs
using System;
using System.Collections.Generic;
using Ducode.Wolk.Domain.Entities.Enums;

namespace Ducode.Wolk.Domain.Entities
{
    public class Note : BaseEntity
    {
        public string Title { get; set; }

        public string Content { get; set; }

        public NoteType NoteType { get; set; }

        public DateTimeOffset? Opened { get; set; }

        public long NotebookId { get; set; }

        public Notebook Notebook { get; set; }

        public ICollection<Attachment> Attachments { get; set; }

        public ICollection<NoteHistory> NoteHistory { get; set; }
    }
}
